pkg/core/native/native_test/neo_test.go

```diff
@@ -117,6 +117,54 @@ func TestNEO_CandidateEvents(t *testing.T) {
 	aer = e.GetTxExecResult(t, tx)
 	require.Equal(t, 0, len(aer.Events))
 }
+func TestNEO_CommitteeEvents(t *testing.T) {
+	neoCommitteeInvoker := newNeoCommitteeClient(t, 100_0000_0000)
+	neoValidatorsInvoker := neoCommitteeInvoker.WithSigners(neoCommitteeInvoker.Validator)
+	e := neoCommitteeInvoker.Executor
+
+	cfg := e.Chain.GetConfig()
+	committeeSize := cfg.GetCommitteeSize(0)
+
+	voters := make([]neotest.Signer, committeeSize)
+	candidates := make([]neotest.Signer, committeeSize)
+	for i := 0; i < committeeSize; i++ {
+		voters[i] = e.NewAccount(t, 10_0000_0000)
+		candidates[i] = e.NewAccount(t, 2000_0000_0000) // enough for one registration
+	}
+	txes := make([]*transaction.Transaction, 0, committeeSize*3)
+	for i := 0; i < committeeSize; i++ {
+		transferTx := neoValidatorsInvoker.PrepareInvoke(t, "transfer", e.Validator.ScriptHash(), voters[i].(neotest.SingleSigner).Account().PrivateKey().GetScriptHash(), int64(committeeSize-i)*1000000, nil)
+		txes = append(txes, transferTx)
+
+		registerTx := neoValidatorsInvoker.WithSigners(candidates[i]).PrepareInvoke(t, "registerCandidate", candidates[i].(neotest.SingleSigner).Account().PublicKey().Bytes())
+		txes = append(txes, registerTx)
+
+		voteTx := neoValidatorsInvoker.WithSigners(voters[i]).PrepareInvoke(t, "vote", voters[i].(neotest.SingleSigner).Account().PrivateKey().GetScriptHash(), candidates[i].(neotest.SingleSigner).Account().PublicKey().Bytes())
+		txes = append(txes, voteTx)
+	}
+	block := neoValidatorsInvoker.AddNewBlock(t, txes...)
+	for _, tx := range txes {
+		e.CheckHalt(t, tx.Hash(), stackitem.Make(true))
+	}
+
+	// Advance the chain to trigger committee recalculation and potential change
+	for (block.Index)%uint32(committeeSize) != 0 {
+		block = neoCommitteeInvoker.AddNewBlock(t)
+	}
+	// Check for CommitteeChanged event in the last persisted block's AER
+	blockHash := e.Chain.CurrentBlockHash()
+	aer, err := e.Chain.GetAppExecResults(blockHash, trigger.OnPersist)
+	require.NoError(t, err)
+	require.NotEmpty(t, aer)
+	found := false
+	for _, event := range aer[0].Events {
+		if event.Name == "CommitteeChanged" {
+			found = true
+			break
+		}
+	}
+	require.True(t, found, "CommitteeChanged event not found")
```

Comment:
It's not enough. The block is known and empty, the order of events is strictly fixed. Thus, instead of this check we need to ensure that the CommitteeChanged event is placed at the correct index and contains exactly the expected stackitems. Old committee is chain's StandBy committee, those public keys may be retrieved from the chain's config, and the new committee should be constructed from the list of registered users.